client/state/signup/progress/actions.js

```diff
@@ -128,3 +129,10 @@ export function invalidateStep( step, errors ) {
 		errors,
 	};
 }
+
+export function removeStep( step ) {
```

Comment:
I may have missed it, but I can't see where this new action is used. I see it's passed into the the main `<Signup>` component, but don't see it get called. Is it still needed?

Reply:
I use removeStep in the `shouldShowUpsell()` function in [this line](https://github.com/Automattic/wp-calypso/pull/40879/files#diff-1236f5fcce1a6ae92a78305706a9d3f7R775). The reason removeStep is needed is that a user could do the following steps:
Select paid plan -> Hit back in domains step -> Select free plan -> Select custom domain.

Selecting a paid plan first will fulfill the `upsell-plan` and `plans-plan-only` steps and add those steps to the progress store as "completed". This is because we don't want to show the upsell for users who've selected a paid plan. Now, going back on the Domains step and selecting the free plan should remove the fulfilled steps from progress store since we might probably show the upsell depending on the path the user takes from here -> if they select a free domain, the steps upsell and second plan steps get fulfilled again; if they select a custom domain, then they see the upsell offer.

Reply:
That makes sense, thanks for explaining that.

The difference between `removeStep` (which removes the dependencies provided by the step) and "excluding a step" (which excludes it from the flow) is pretty subtle.

What do you think about calling it `unsubmitSignupStep()` since it's sort of goes hand in hand with `submitSignupStep()`? Or have I still misunderstood the purpose? 😅

Reply:
Agree. Going by the naming style followed in the [reducer file](https://github.com/Automattic/wp-calypso/blob/786ca0e353f64ccfd8d87fb538bbbbad1aca13a6/client/state/signup/progress/reducer.js#L111), I think unsubmitStep would work well. I'll do that 👍